Application/ColourSurface.pde

```diff
@@ -0,0 +1,82 @@
+class ColourSurface extends VectorSurface{
+  ArrayList<Gradient> gradientMakers;
+  
+  ColourSurface(){
+    super();
+    gradientMakers = new ArrayList<Gradient>();
```

Comment:
extract this copied code to private constructor and invoke it (like in http://stackoverflow.com/questions/2373988/call-one-constructor-from-another-in-java)